HARK/ConsumptionSaving/ConsMarkovModel.py

```diff
@@ -782,6 +783,32 @@ def initializeSim(self):
             Cutoffs = np.cumsum(np.array(self.MrkvPrbsInit))
             self.MrkvNow = np.ones(self.AgentCount)*np.searchsorted(Cutoffs,base_draw).astype(int)
         self.MrkvNow = self.MrkvNow.astype(int)
+        
+        
+    def resetRNG(self):
+        '''
+        Extended method that ensures random shocks are drawn from the same sequence
+        on each simulation, which is important for structural estimation.  This
+        method is called automatically by initializeSim().
+        
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+        '''
+        PerfForesightConsumerType.resetRNG(self)
+        
+        # Reset IncomeDstn if it exists (it might not because resetRNG is called at init)
```

Comment:
I wonder if this could be written in a more general way once rather than having more complexity added via inheritance.

This could be a time to think through how exogenous shocks get stored on the Agent object.

dolo, for example, puts all these into a collection names `exogenous`